server/src/main/java/org/tctalent/server/api/admin/PartnerAndRoleBasedDtoPropertyFilter.java

```diff
@@ -27,162 +27,157 @@
-     * @param fullyVisibleCandidates If present, this specifies the candidates that we can return
-     *                               all data for (ie no filtering)
-     * @param publicProperties Names of properties that are always returned (ie never filtered out)
-     * @param semiLimitedExtraProperties Names of properties that may be filtered out.
-     */
-    public PartnerAndRoleBasedDtoPropertyFilter(@Nullable Partner partner, @NonNull Role role,
-        @Nullable Set<Candidate> fullyVisibleCandidates,
-        @Nullable Set<String> publicProperties, @Nullable Set<String> semiLimitedExtraProperties) {
-        this.partner = partner;
-        this.role = role;
-        this.publicProperties = publicProperties;
-        this.semiLimitedExtraProperties = semiLimitedExtraProperties;
-
-        setFullyVisibleCandidates(fullyVisibleCandidates);
-    }
-
-    @Override
-    public boolean ignoreProperty(Object o, String property) {
-        boolean ignore;
-
-        if (role == Role.systemadmin
-            //Allows default partner (eg TBB) admins to see everything.
-            || isDefaultPartner(partner) && (role == Role.admin || role == Role.partneradmin)
-            || isInFullyVisibleCandidates(o)
-        ) {
-            ignore = false;
+  @Nullable
+  private Set<Candidate> fullyVisibleCandidates;
+  @Nullable
+  private Set<User> fullyVisibleUsers; //Computed from fullyVisibleCandidates
+  @Nullable
+  private final Partner partner;
+
+  @NonNull
+  private final Role role;
+
+  @Nullable
+  private final Set<String> publicProperties;
+
+  @Nullable
+  private final Set<String> semiLimitedExtraProperties;
+
+  /**
+   * Filters out non-public Candidate properties according the given parameters
+   *
+   * @param partner                    Partner associated with the user requesting the candidate
+   *                                   data.
+   * @param role                       Role of user requesting the candidate data
+   * @param fullyVisibleCandidates     If present, this specifies the candidates that we can return
+   *                                   all data for (ie no filtering)
+   * @param publicProperties           Names of properties that are always returned (ie never
+   *                                   filtered out)
+   * @param semiLimitedExtraProperties Names of properties that may be filtered out.
+   */
+  public PartnerAndRoleBasedDtoPropertyFilter(@Nullable Partner partner, @NonNull Role role,
+      @Nullable Set<Candidate> fullyVisibleCandidates,
+      @Nullable Set<String> publicProperties, @Nullable Set<String> semiLimitedExtraProperties) {
+    this.partner = partner;
+    this.role = role;
+    this.publicProperties = publicProperties;
+    this.semiLimitedExtraProperties = semiLimitedExtraProperties;
+
+    setFullyVisibleCandidates(fullyVisibleCandidates);
+  }
+
+  @Override
+  public boolean ignoreProperty(Object o, String property) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 12. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.